lib/Geo/ShapeContainer.cpp

```diff
@@ -676,20 +673,24 @@ bool ShapeContainer::intersects(S2Polyline const* other) const {
 }
 
 namespace {
+
 bool intersectRectPolygon(S2LatLngRect const* rect, S2Polygon const* poly) {
   // only used in legacy situations
-  if (rect->is_full()) {
-    return true;  // rectangle spans entire sphere
-  } else if (rect->is_point()) {
-    return poly->Contains(rect->lo().ToPoint());  // easy case
-  } else if (!rect->Intersects(poly->GetRectBound())) {
-    return false;  // cheap rejection
+  if (ADB_UNLIKELY(rect->is_point())) {
```

Comment:
NIT 
Why is this "unlikely" and others are "likely"?  I assume it depends on the data in the collection.